libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts

```diff
@@ -132,11 +139,48 @@ export class BeginSkill extends SkillDialog {
         return await super.continueDialog(dc);
     }
 
+    public async repromptDialog(turnContext: TurnContext, instance: DialogInstance): Promise<void> {
+        this.loadDialogOptions(turnContext, instance);
+        return await super.repromptDialog(turnContext, instance);
+    }
+
+    public async resumeDialog(dc: DialogContext, reason: DialogReason, result?: any): Promise<DialogTurnResult<any>> {
+        this.loadDialogOptions(dc.context, dc.activeDialog);
+        return await super.resumeDialog(dc, reason, result);
+    }
+
+    public async endDialog(turnContext: TurnContext, instance: DialogInstance, reason: DialogReason): Promise<void> {
+        this.loadDialogOptions(turnContext, instance);
+        return await super.endDialog(turnContext, instance, reason);
+    }
+
     protected onComputeId(): string {
         const appId = this.skillAppId ? this.skillAppId.toString() : '';
         if (this.activity instanceof ActivityTemplate) {
             return `BeginSkill['${ appId }','${ StringUtils.ellipsis(this.activity.template.trim(), 30) }']`;
         }
         return `BeginSkill['${ appId }','${ StringUtils.ellipsis(this.activity && this.activity.toString().trim(), 30) }']`;
     }
+
+    private loadDialogOptions(context: TurnContext, instance: DialogInstance): void {
+        const dialogOptions = <SkillDialogOptions>instance.state[this._dialogOptionsStateKey];
+
+        this.dialogOptions.botId = dialogOptions.botId;
+        this.dialogOptions.skillHostEndpoint = dialogOptions.skillHostEndpoint;
+        
+        this.dialogOptions.conversationIdFactory = context.turnState.get(skillConversationIdFactoryKey);
+        if (this.dialogOptions.conversationIdFactory == null) { 
+            throw new ReferenceError('Unable to locate skillConversationIdFactoryBase in HostContext.');
+        };
+        
+        this.dialogOptions.skillClient = context.turnState.get(skillClientKey);
```

Comment:
prevent serialization

Reply:
This shouldn't be an issue as we don't prevent serialization of anything else in the JS SDK like we do in the C# SDK. Something to watch out for, though.

Reply:
#Resolved for now